recipes-bsp/acpica/files/manipulate-fds-instead-of-FILE.patch

```diff
@@ -0,0 +1,77 @@
+From 48bba6c36dc238d1fabed582d1891e947a667a4e Mon Sep 17 00:00:00 2001
+From: Patrick Ohly <patrick.ohly@intel.com>
+Date: Thu, 23 Feb 2017 18:10:47 +0100
+Subject: [PATCH] aslfiles.c: manipulate fds instead of FILE
+
+Copying what stdout/stderr point to is not portable and fails with
+musl because FILE is an undefined struct.
+
+Instead, use lower-level Unix functions to modify the file that stderr
+writes into. This works on the platforms that Yocto targets.
+
+Upstream-Status: Inappropriate [embedded specific]
+
+Signed-off-by: Patrick Ohly <patrick.ohly@intel.com>
+Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>
+---
+ source/compiler/aslfiles.c | 28 ++++++++++++++--------------
+ 1 file changed, 14 insertions(+), 14 deletions(-)
+
+diff --git a/source/compiler/aslfiles.c b/source/compiler/aslfiles.c
+index 82865db..114345f 100644
+--- a/source/compiler/aslfiles.c
++++ b/source/compiler/aslfiles.c
+@@ -44,6 +44,11 @@
+ #include "aslcompiler.h"
+ #include "acapps.h"
+ 
++#include <sys/types.h>
++#include <sys/stat.h>
++#include <fcntl.h>
++#include <unistd.h>
++
+ #define _COMPONENT          ACPI_COMPILER
+         ACPI_MODULE_NAME    ("aslfiles")
+ 
+@@ -606,6 +611,8 @@ FlOpenMiscOutputFiles (
+ 
+     if (Gbl_DebugFlag)
+     {
++        int fd;
++
+         Filename = FlGenerateFilename (FilenamePrefix, FILE_SUFFIX_DEBUG);
+         if (!Filename)
+         {
+@@ -617,22 +624,15 @@ FlOpenMiscOutputFiles (
+         /* Open the debug file as STDERR, text mode */
+ 
+         Gbl_Files[ASL_FILE_DEBUG_OUTPUT].Filename = Filename;
+-        Gbl_Files[ASL_FILE_DEBUG_OUTPUT].Handle =
+-            freopen (Filename, "w+t", stderr);
+-
+-        if (!Gbl_Files[ASL_FILE_DEBUG_OUTPUT].Handle)
++        fd = open(Filename, O_CREAT|O_TRUNC, S_IRUSR|S_IWUSR|S_IRGRP|S_IWGRP|S_IROTH|S_IWOTH);
++        if (fd < 0 ||
++            dup2(fd, fileno(stderr)))
+         {
+-            /*
+-             * A problem with freopen is that on error, we no longer
+-             * have stderr and cannot emit normal error messages.
+-             * Emit error to stdout, close files, and exit.
+-             */
+-            fprintf (stdout,
+-                "\nCould not open debug output file: %s\n\n", Filename);
+-
+-            CmCleanupAndExit ();
+-            exit (1);
```

Comment:
I am mostly worried about this chunk, as I got the conflict here. I selected the solution from the old acpica patch, but clearly upstream had also been making changes in the same area to fix the same issue, so maybe this chunk should have been dropped.

Reply:
Can we just try new ACPICA version w/o any additional 3rd party patches?